Added a feature to backup on saving the TiddlyWiki. …

oneseedfruit opened the pull request:
I think #1 backup on saving is important.

<a href="https://ibb.co/huJSRm"><img src="https://preview.ibb.co/nBfwK6/Screenshot_2018_01_11_12_26_00.png" alt="Screenshot_2018_01_11_12_26_00" border="0"></a>

(I used Visual Studio 2017 and I also threw in a different version of ObjectListView while working on it. Please check and fix them should there be incompatibilities.)

oneseedfruit committed:
Added a feature to backup on saving the TiddlyWiki. I used Visual Studio 2017 and I also threw in a different version of ObjectListView while working on it. Please check and fix them should there be incompatibilities.

lamusia commented:
Thank you for the idea. But the patch can’t be merged because:
1.	If you need a failsafe backup, then it’s unnecessary. ATWiki is unlike web server may vulnerable to network issue. And during the test, not even once data lost happened.
2.	For history version backup, there are so many option. Like windows built-in file history function. Or cloud backup like OneDrive.
3.	The backup method you provide is not general purpose. ATWiki should be simple and easy to use, no redundancy function, no reinvent the wheel.
4.	Backup is important. Let user decide their method.

However, you can still use your modified version.

oneseedfruit commented:
I'm not concerned about failsafe backup or history version backup. 

My main concern is that when using ATWiki, if I modified anything in the TiddlyWiki, then save it, it will just overwrite the file. If I accidentally deleted a tiddler and saved it, then it'll be gone just like that because ATWiki will just overwrite the file.

What I added is, if I modify the wiki and save it, it will create and extra copy of the same TiddlyWiki html file and put it in the designated directory, right upon saving. This is similar to the backup feature in [TiddlyDesktop](https://github.com/Jermolene/TiddlyDesktop).

lamusia commented:
Why don’t you use a git/svn repository to store the documents? After editing, commit with one more check. This should prevent any accidental content delete.

oneseedfruit commented:
But that would make ATWiki unsuited to be used on its own as a quick note-taking client as it requires 2 separate software to be used safely, not to mention the need to maintain a git repository.

And it is also a hassle to take notes quickly, since every save needs to be done in 2 steps:
1. saving the wiki in ATWiki
2. git commit

Also, the need to maintain a git repository just for it.

So every change no matter how big or small, or even the really trivial ones like correcting a spelling error, would require 2 steps. 

Aside from TiddlyDesktop, [TiddlyServer](https://github.com/Arlen22/TiddlyServer) too, creates a copy of the original file as backup upon saving.

lamusia commented:
That’s reasonable. Working on it.
